website/project/model.py

```diff
@@ -4588,3 +4622,36 @@ def approve(self, user):
     def reject(self, user):
         self.approval.reject(user)
         self.approval.save()
+
+
+@unique_on(['destination_node', '_id'])
+class MailingListEventLog(StoredObject):
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    date_created = fields.DateTimeField(default=datetime.datetime.utcnow, index=True)
+
+    content = fields.StringField()
+    sending_email = fields.StringField(index=True)
+    sending_user = fields.ForeignField('user', index=True)
+    destination_node = fields.ForeignField('node', index=True)
+    status = fields.StringField()
+
+    # Possible statuses
+    UNAUTHORIZED = 'no_user'
+    NOT_FOUND = 'node_dne'
+    DELETED = 'node_deleted'
+    FORBIDDEN = 'no_access'
+    DISABLED = 'discussions_disabled'
+    OK = 'sent'
+
+    @classmethod
+    def create_from_event(cls, content, status, node, email, user):
```

Comment:
(I don't find a constructor method on [`NodeLog`](https://github.com/CenterForOpenScience/osf.io/blob/e16b8948506548011e7e617af32ffffc9428501a/website/project/model.py#L385-L585), which is otherwise similar.)